### PR TITLE
Update to Eclipse 2024-09 and Java 21

### PR DESCRIPTION
**Pull request by buchen**
This change
* updates to Eclipse 2024-09
* updates dependencies to their respective latest version
* updates to Java 21
* update the embedded JRE (i.e. the distro is not launching because it still includes Java 17)

### Test Launch
- [x] Linux x86_64
- [x] Linux aarch64
- [x] macOS x86_64
- [x] macOS aarch64
- [x] Windows x86_64

### Test update from 0.70.4 to Java21-based installation
- [x] Linux x86_64
- [x] Linux aarch64
- [x] macOS x86_64
- [x] macOS aarch64
- [x] Windows x86_64

### Test

Update the update URL in the settings to point to:
```
https://updates.portfolio-performance.info/portfolio-java21
```

⚠️ Do not test on your "productive" installation. Unzip/install a copy to a separate folder.

### CODE REVIEW

**Comment by buchen** [comment]
@Nirus2000 If you happen to have time, can you test the windows update? (On a copy / separate install)

The basic steps should work
* update the update repository URL in the settings
* choose Help -> Search for updates
* there are no release notes yet (empty window expected)
* install
* restart
* check the "installation details" in the "About Portfolio Performance" dialog - is it java21?

**Comment by Nirus2000** [comment]
Start testing ... 

- [ ] Installing [21.0.4+7 Azul Zulu: 21.36.17](https://cdn.azul.com/zulu/bin/zulu21.36.17-ca-jdk21.0.4-win_x64.msi)
- [ ]

**Comment by buchen** [comment]
Thanks. The ```requiredJavaVersion``` is fixed, but not published in the test update site. Thanks.

**Comment by buchen** [comment]
@Nirus2000 

I think you tested the installation package without the embedded Java Runtime Environment.

```
java.home: C:\Program Files\Zulu\zulu-21

...

json-path (2.9.0)
name.abuchen.portfolio (0.70.5.202409151933) [signed by Andreas Buchen] [trusted]
name.abuchen.portfolio.bootstrap (0.70.5.202409151933) [signed by Andreas Buchen] [trusted]
name.abuchen.portfolio.ui (0.70.5.202409151933) [signed by Andreas Buchen] [trusted]
net.minidev.accessors-smart (2.5.1)
```

Can you test a "distro", i.e. a version that includes the JRE 17. And then is updated in place to Java 21.

You can take the [installer](https://github.com/portfolio-performance/portfolio/releases/tag/0.70.4) published, or you build the current destro locally ```mvn clean verify -Ppackage-distro```

**Comment by Nirus2000** [comment]
@buchen 
I think I did it this way. 

### With JDK17
grafik

---

### With JDK21
grafik

- Downloaded ZIP
- Opened test file
- Changed update site in settings and saved
- Change the JDK to Java 21
- Restarted and started download for dependencies

---

### Fast Switch JDK Version
[scripts.zip](https://github.com/user-attachments/files/17020020/scripts.zip)

- Copy this folder to `"C:\Program Files\Zulu\"`
- Change in Environment
  - Add
grafik
- Delete all other old JDK entry

Look in in the *.bat Files for the logic:
grafik


### *BATCH FILES ROCK's* 😆🤘🤘🤘

Everything worked.

**Comment by buchen** [comment]
> BATCH FILES ROCK

😄

Fair. But what I want to test is that the **self-update** of the Java Runtime Environment works on Windows. If users have installed using the Windows installer, then the JRE 17 is embedded in the installation and must be updated to Java 21 as well. I tested it on macOS and Linux (both with ARM), but I miss a testing environment on Windows.

File: portfolio-product/name.abuchen.portfolio.product [hunk 1]
```diff
@@ -62,20 +67,27 @@
...
       <property name="osgi.instance.area.default" value="@user.home/.PortfolioPerformance/workspace" os="linux" />
-      <property name="osgi.requiredJavaVersion" value="17" />
+      <property name="osgi.requiredJavaVersion" value="21" />
       <property name="org.eclipse.update.reconcile" value="false" />
       <property name="eclipse.buildId" value="${unqualifiedVersion}.${buildQualifier}" />
    </configurations>
 
+   <preferencesInfo>
+      <targetfile overwrite="false"/>
+   </preferencesInfo>
```
Comment (buchen) [inline portfolio-product/name.abuchen.portfolio.product:88]: What does this property do?

Comment (Nirus2000) [reply portfolio-product/name.abuchen.portfolio.product:88]: This was automatic added by Eclipse.
We can remove it ...

Comment (buchen) [reply portfolio-product/name.abuchen.portfolio.product]: I have no opinion yet. I do not know what this property does.

Comment (buchen) [reply portfolio-product/name.abuchen.portfolio.product]: I checked. We are not using additional plugin configuration. And even if we used it, we would want to not overwrite it. So I think it is safe to keep it there (and thereby keep the file up-to-date with the latest Eclipse framework).

**Pull request merged**